ctx.go

```diff
@@ -1398,47 +1399,99 @@ func (c *DefaultCtx) Send(body []byte) error {
 	return nil
 }
 
-var (
-	sendFileOnce    sync.Once
-	sendFileFS      *fasthttp.FS
-	sendFileHandler fasthttp.RequestHandler
-)
+// SendFile defines configuration options when to transfer file with SendFileWithConfig.
+type SendFile struct {
+	// FS is the file system to serve the static files from.
+	// You can use interfaces compatible with fs.FS like embed.FS, os.DirFS etc.
+	//
+	// Optional. Default: nil
+	FS fs.FS
+
+	// When set to true, the server tries minimizing CPU usage by caching compressed files.
+	// This works differently than the github.com/gofiber/compression middleware.
+	// Optional. Default value false
+	Compress bool `json:"compress"`
+
+	// When set to true, enables byte range requests.
+	// Optional. Default value false
+	ByteRange bool `json:"byte_range"`
+
+	// When set to true, enables direct download.
+	//
+	// Optional. Default: false.
+	Download bool `json:"download"`
+
+	// Expiration duration for inactive file handlers.
+	// Use a negative time.Duration to disable it.
+	//
+	// Optional. Default value 10 * time.Second.
+	CacheDuration time.Duration `json:"cache_duration"`
+
+	// The value for the Cache-Control HTTP-header
+	// that is set on the file response. MaxAge is defined in seconds.
+	//
+	// Optional. Default value 0.
+	MaxAge int `json:"max_age"`
+}
 
 // SendFile transfers the file from the given path.
 // The file is not compressed by default, enable this by passing a 'true' argument
 // Sets the Content-Type response HTTP header field based on the filenames extension.
-func (c *DefaultCtx) SendFile(file string, compress ...bool) error {
+func (c *DefaultCtx) SendFile(file string, config ...SendFile) error {
 	// Save the filename, we will need it in the error message if the file isn't found
 	filename := file
 
+	route := c.Route()
```

Comment:
not sure if we should store this data in the route itself, if then abstracted 
what if you have multiple sendFiles in a route ? 
shouldn't we create a map with the filenames and config settings as a key and store it underneath ?

Reply:
> not sure if we should store this data in the route itself, if then abstracted what if you have multiple sendFiles in a route ? shouldn't we create a map with the filenames and config settings as a key and store it underneath ?

The filename itself is not an issue since we don't keep it anywhere. 

Where can we store FS other than routes? I don't think it's a good idea to store them with filename since the filename can be changed according to the request

Reply:
@efectn we can create a separate map and use the filename and the config as key
we don't want to extend the route for the feature, otherwise we can't abstract it later and it has several responsibilities that should be separated.